File: packages/graph/__fixtures__/multi-lang-project/src/services/AuthService.java
package com.example.services;

import com.example.types.User;
import com.example.types.AuthToken;
import com.example.utils.HashUtil;

public class AuthService {
    private String secret;

    public AuthService(String secret) {
        this.secret = secret;
    }

    public AuthToken authenticate(String username, String password) {
        String hashed = HashUtil.hashPassword(password);
        return new AuthToken(hashed, username);
    }

    public boolean validateToken(String token) {
        return token != null && !token.isEmpty();
    }

    private String getSecret() {
        return this.secret;
    }
}

File: packages/graph/__fixtures__/multi-lang-project/src/services/AuthService.rs
use crate::types::{User, AuthToken};
use crate::utils::hash::hash_password;

pub struct AuthService {
    secret: String,
}

pub fn create_auth_service(secret: String) -> AuthService {
    AuthService { secret }
}

impl AuthService {
    pub fn authenticate(&self, username: &str, password: &str) -> AuthToken {
        let hashed = hash_password(password);
        AuthToken {
            token: hashed,
            user: username.to_string(),
        }
    }

    pub fn validate_token(&self, token: &str) -> bool {
        !token.is_empty()
    }
}

pub const MAX_SESSIONS: u32 = 100;

pub trait Authenticator {
    fn login(&self, username: &str, password: &str) -> bool;
}
